src/v/archival/ntp_archiver_service.cc

```diff
@@ -1387,18 +1389,18 @@ ntp_archiver::schedule_uploads(std::vector<upload_context> loop_contexts) {
         std::exception_ptr ep;
         try {
             while (uploads_remaining > 0 && may_begin_uploads()) {
-                auto should_stop = co_await ctx.schedule_single_upload(*this);
-                if (should_stop == ss::stop_iteration::yes) {
+                auto scheduled = co_await schedule_single_upload(ctx);
+                ctx.start_offset = scheduled.inclusive_last_offset
+                                   + model::offset(1);
```

Comment:
nit: this could be a `model::next_offset(scheduled.inclusive_last_offset)`

Reply:
I'll split it into another commit for now.